vllm/v1/spec_decode/eagle.py

```diff
@@ -334,6 +334,14 @@ def load_model(self, target_model: nn.Module) -> None:
             logger.info("Loading EAGLE LM head weights from the target model.")
             self.model.lm_head = target_model.lm_head
 
+        if self.vllm_config.speculative_config.method == "eagle3" and \
+                hasattr(target_model, "draft_id_to_target_id") and \
+                "d2t" not in loaded_weights:
+            logger.info("Draft id to target id mapping not found. Assuming "
+                        "draft and target model share the same vocab.")
+            del self.model.draft_id_to_target_id
+            self.model.draft_id_to_target_id = None
```

Comment:
Why not fix this in eagle3 model?

Reply:
Good point. I hadn't noticed that all the other logic in `eagle.py` exists to copy data from the target model, which is not needed here. Changed.